### ISSUE TITLE
Outflow of reservoir/lake

### ISSUE THREAD

**Issue by hboisgon** [Deltares/Wflow.jl#267]
Maybe a little related to #90 but I have a question on how the reservoir/lake outflow are being given back to the wflow model.
What I see from the code is that the outflow is added into qin of the downstream cell rather than directly changing the q/q_av value of the cell itself (see few line below):

https://github.com/Deltares/Wflow.jl/blob/e3de2bc6d9a50a04e5936bff33ce7e73fe921d55/src/flow.jl#L284-L298

The thing is that then for the reservoir/lake outlet cell, if you save q_av you get a completely hypothetic value and the outflow of the lake is masked as an external inflow to the downstream cell instead. So if you would make a mass balance over the surface flow cells as a postprocess (eg coupling to Delwaq) then you have to manually overwrite the volume and q_av of the lake outlet to get the mass balances right. I think it would be easier and maybe more intuitive if for these cells both q/q_av/volume would be overwritten by the lake/reservoir data ?
This would also fix the problem of modelling a lake without a downstream node.

**Comment by hboisgon** [comment]
Thanks for your quick reply @verseve !
I agree that for volume then this indeed can be kept to what it is.

I'm not sure it happens so often with the classic resolution of wflow that the lake/reservoir area has a much smaller resolution than the cell itself? I mean qlat and q from upstream would already be considered so the lake would have to be really small to get significant errors (1/3 of the cell or less?). The global lake model is anyhow only good for large lakes (>10km2)

**Comment by verseve** [comment]
> I'm not sure it happens so often with the classic resolution of wflow that the lake/reservoir area has a much smaller resolution than the cell itself? I mean qlat and q from upstream would already be considered so the lake would have to be really small to get significant errors (1/3 of the cell or less?). The global lake model is anyhow only good for large lakes (>10km2)

Yes, I agree, for the `SBM` concept, where we often run with a ~1 km resolution (default) this is not an issue I would say. I am not sure about the model resolution of other concepts like `FLEXTopo` (@laurenebouaziz  ?) or `HBV`, it is probably close to `SBM`. Anyway, if we change this, we can also explain this in more detail in the documentation.
 
> In a way you could also still run the kinematic wave first but then instead of adding the lake outflow to the dowsntream overwrite the q and q_av from the intermediate kinematic wave instead?

I am not sure if I understand this question completely. I think you propose to overwrite `q` and `q_av` of the kinematic wave in the same cell as the reservoir/lake outlet location with the reservoir/lake outflow? This is indeed possible, but for the inflow (from the river) to the reservoir/lake we then have to take the upstream cell(s). Otherwise, the reservoir/lake outflow is also used as inflow. Or, we have to account for both in one cell, not sure if it is worth to go in that direction.

**Comment by laurenebouaziz** [comment]
yes I think we can in general assume the flextopo resolution to be similar to the sbm resolution (and same for hbv I would say)

**Comment by hboisgon** [comment]
> Otherwise, the reservoir/lake outflow is also used as inflow.

Oh yes indeed, I forgot that part, so ignore what I said, then just indeed overwrite ``q`` and ``q_av``
